Assets/Scripts/Player/PlayerControls.cs

```diff
@@ -120,6 +179,7 @@ public override void Update()
 			else
```

Comment:
When checking if we should climb before this, it's cheaper to do ClimbCheck last since you could early out if holdbutton isn't down or if we can't use stamina.